benchmark/infrastructure/BenchmarkToString.cpp

```diff
@@ -0,0 +1,149 @@
+ * "#################
+ * # categoryTitle #
+ * #################"
+ */
+static std::string createCategoryTitle(std::string_view categoryTitle) {
+  // The bar above and below the title.
+  const size_t barLength = categoryTitle.size() + 4;
+  const std::string bar(barLength, '#');
+
+  return absl::StrCat(bar, "\n# ", categoryTitle, " #\n", bar);
+}
+
+// ___________________________________________________________________________
+std::string addIndentation(const std::string_view str,
+                           const size_t& indentationLevel) {
+  // An indention level of 0 makes no sense. Must be an error.
+  AD_CONTRACT_CHECK(indentationLevel > 0);
+
+  // The indention symbols for this level of indention.
+  std::string indentationSymbols{""};
+  indentationSymbols.reserve(outputIndentation.size() * indentationLevel);
+  for (size_t i = 0; i < indentationLevel; i++) {
+    indentationSymbols.append(outputIndentation);
+  }
+
+  // Add an indentation to the beginning and replace a new line with a new line,
+  // directly followed by the indentation.
+  return absl::StrCat(
+      indentationSymbols,
+      absl::StrReplaceAll(str,
+                          {{"\n", absl::StrCat("\n", indentationSymbols)}}));
+}
+
+// ___________________________________________________________________________
+std::string getMetadataPrettyString(const BenchmarkMetadata& meta,
+                                    std::string_view prefix,
+                                    std::string_view suffix) {
+  const std::string& metadataString = meta.asJsonString(true);
+  if (metadataString != "null") {
+    return absl::StrCat(prefix, metadataString, suffix);
+  } else {
+    return "";
+  }
+}
+
+/*
+@brief Return a string containing a titel version of `categoryName` and a string
+list representation of all the given category entries.
+Note: This function converts `CategoryType` objects by trying to cast them as
+`std::string`.
+*/
+template <typename CategoryType>
+static std::string categoryToString(
+    std::string_view categoryName,
+    const std::vector<CategoryType>& categoryEntries) {
+  return absl::StrCat(createCategoryTitle(categoryName), "\n",
+                      ad_utility::listToString(
+                          categoryEntries,
+                          [](const CategoryType& entry) {
+                            return static_cast<std::string>(entry);
+                          },
+                          "\n\n"));
+}
+
+// ___________________________________________________________________________
+std::string benchmarkResultsToString(
+    const BenchmarkInterface* const benchmarkClass,
+    const BenchmarkResults& results) {
+  // The values for all the categories of benchmarks.
+  const std::vector<ResultEntry>& singleMeasurements =
+      results.getSingleMeasurements();
+  const std::vector<ResultGroup>& resultGroups = results.getGroups();
+  const std::vector<ResultTable>& resultTables = results.getTables();
```

Comment:
These variables are only used exactly once, and much further down.
Just replace them by their respective functions calls and save those three lines.

Reply:
Got point. Must have been an artifact. I got quite a lot of those lying around... ([86118c7](https://github.com/ad-freiburg/qlever/pull/1010/commits/86118c721e122bdd346cf93a080963b68a0427fb))